Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/Instance.cpp

```diff
@@ -223,10 +223,10 @@ namespace AzToolsFramework
                 }
             }
 
-            return DetachEntity(entityAliasToRemove);
+            return EraseEntityFromMap(entityAliasToRemove);
         }
 
-        AZStd::unique_ptr<AZ::Entity> Instance::DetachEntity(const EntityAlias& entityAlias)
+        AZStd::unique_ptr<AZ::Entity> Instance::EraseEntityFromMap(const EntityAlias& entityAlias)
```

Comment:
I don't think 'EraseEntityFromMap' is a better name than DetachEntity(). Detach implies that it removes and returns entity for further action. Erase doesn't imply that. I don't think having the same name for public and private functions is an issue because it's only the public one that's accessible. If you still prefer a different name, you can go with DetachEntityHelper()